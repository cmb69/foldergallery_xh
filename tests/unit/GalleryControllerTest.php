<?php

/**
 * Copyright 2023 Christoph M. Becker
 *
 * This file is part of Foldergallery_XH.
 *
 * Foldergallery_XH is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Foldergallery_XH is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Foldergallery_XH.  If not, see <http://www.gnu.org/licenses/>.
 */

namespace Foldergallery;

use ApprovalTests\Approvals;
use Foldergallery\Infra\ImageService;
use Foldergallery\Infra\View;
use PHPUnit\Framework\TestCase;

class GalleryControllerTest extends TestCase
{
    public function testRendersEmptyGallery(): void
    {
        $sut = new GalleryController(
            "./plugins/foldergallery/",
            $this->conf(),
            $this->text(),
            $this->imageService(),
            $this->view()
        );
        $response = $sut("/?Gallery", "test");
        Approvals::verifyHtml($response);
    }

    private function imageService(): ImageService
    {
        $service = $this->createMock(ImageService::class);
        $service->method("findEntries")->willReturn([]);
        return $service;
    }

    private function view(): View
    {
        return new View("./views/", $this->text());
    }

    private function conf(): array
    {
        return XH_includeVar("./config/config.php", "plugin_cf")["foldergallery"];
    }

    private function text(): array
    {
        return XH_includeVar("./languages/en.php", "plugin_tx")["foldergallery"];
    }
}
