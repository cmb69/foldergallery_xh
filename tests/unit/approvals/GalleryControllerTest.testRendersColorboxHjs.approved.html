<link rel="stylesheet" href="./plugins/foldergallery/colorbox/colorbox.css" type="text/css">
